pandas/core/groupby/groupby.py

```diff
@@ -1837,14 +1837,24 @@ def mean(
         Name: B, dtype: float64
         """
 
+        if not (self.obj.mean is Series.mean or self.obj.mean is DataFrame.mean):
```

Comment:
Checking for method equality breaks lots of the tests in `test_groupby.py`, I am still working on why this is

Reply:
@jorisvandenbossche would you expect this comparison to work? I've never compared methods in this way